common/math.go

```diff
@@ -0,0 +1,27 @@
+package common
+
+import (
+	"fmt"
+)
+
+func Complement[T comparable](slice []T, toRemove []T) []T {
+	var complement []T
+
+	//put toRemove into a slice in a map for faster lookup
+	removeMap := make(map[T]bool)
+	for _, remove := range toRemove {
+		removeMap[remove] = true
+	}
+
+	for _, element := range slice {
+		_, found := removeMap[element]
+		if !found {
+			complement = append(complement, element)
+		} else {
+			fmt.Printf("\nRemoving: '%v' from slice", element)
```

Comment:
This appeared when I tested locally - do we want to make the message a bit more user friendly, or take it out?

Reply:
I'm not sure there's a way to make it more user friendly, as it's used in multiple places so is deliberately quit general. I could make it a debug log, so it'll only show up if you're looking for it?